cmd/notation/plugin.go

```diff
@@ -62,3 +109,160 @@ func listPlugins(command *cobra.Command) error {
 	}
 	return tw.Flush()
 }
+
+func installPlugin(command *cobra.Command, args []string, force bool) error {
+	if len(args) != 1 {
+		return errors.New("missing plugin package")
+	}
+
+	pluginSrcPath := args[0]
+
+	pluginBinary := filepath.Base(pluginSrcPath)
+
+	// get plugin metadata
+	cmd := exec.Command(pluginSrcPath, "get-plugin-metadata")
+
+	output, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	var newPlugin map[string]interface{}
+	err = json.Unmarshal([]byte(output), &newPlugin)
+	if err != nil {
+		return err
+	}
+
+	pluginName := newPlugin["name"].(string)
+	newPluginVersion := newPlugin["version"].(string)
+	newSemVersion, err := semver.NewVersion(newPluginVersion)
+	if err != nil {
+		return err
+	}
+
+	// get plugin directory
+	pluginDir, err := dir.PluginFS().SysPath(pluginName)
+	if err != nil {
+		return err
+	}
+
+	// Check if plugin directory exists
+	_, err = os.Stat(pluginDir)
+
+	// create the directory, if not exist
+	if os.IsNotExist(err) {
+		if err := os.MkdirAll(pluginDir, 0755); err != nil {
+			return err
+		}
+	}
+
+	pluginDestPath := pluginDir + "/" + pluginBinary
+
+	// Check if plugin exists
+	_, err = os.Stat(pluginDestPath)
+
+	// copy plugin, if not exist
+	if os.IsNotExist(err) {
+		copyPlugin(pluginSrcPath, pluginDestPath)
+	}
```

Comment:
Those lines may can be removed by using `CLIPlugin` and call [GetMetadata](https://github.com/notaryproject/notation/blob/26187094a03643e0b283d40f34e39cf599df75dd/cmd/notation/plugin.go#L128)